### PR TITLE
fix: check seller only can withdraw after the loan funded

### PR DESCRIPTION
**Pull request by Jung0x12**
# Summary
Lack of Validation for Loan Funding Status

Issue: https://www.notion.so/islelabs/Smart-Contract-Audit-Draft-Response-v1-0-03839677f03540238c0d167fdeee2e7e?pvs=97#921949e3b5534f28b05d45b0ca3f3039

# Description
The withdrawFunds function does not validate whether the loan has been fully funded before allowing the seller to withdraw funds. The situation could result in the Seller withdrawing 0 amount but having already transferred the Receivable to LoanManager.

# Fix
Check the drawable funds is not 0 (be funded) before seller can withdraw.

# Verification
- Run tests
   `test_RevertWhen_LoanNotFunded`

### CODE REVIEW

File: tests/Base.t.sol [hunk 1]
```diff
@@ -270,18 +270,20 @@ abstract contract Base_Test is StdCheats, Events, Constants, Utils {
...
             address(receivable),
-            receivablesTokenId_,
+            tokenId_,
             defaults.GRACE_PERIOD(),
             defaults.PRINCIPAL_REQUESTED(),
             [defaults.INTEREST_RATE(), defaults.LATE_INTEREST_PREMIUM_RATE()]
         );
+    }
+
+    function createDefaultLoan() internal {
```
Comment (LouisTsai-Csie) [inline tests/Base.t.sol:284]: nit: The original function logic performs the following operations:

1. Create a default receivable.
2. Request the loan based on the receivable.
3. Fund the loan.

After splitting the function into `requestDefaultLoan` and `createDefaultLoan`, the latter function now only funds the loan. For better readability, I suggest renaming `createDefaultLoan` to `fundDefaultLoan`.

File: tests/integration/concrete/loan-manager/withdraw-funds/withdrawFunds.t.sol [hunk 2]
```diff
@@ -81,6 +84,17 @@ contract WithdrawFunds_LoanManager_Integration_Concrete_Test is
         IERC721(address(receivable)).ownerOf(receivableTokenId);
     }
 
+    modifier whenLoanRequested() {
+        uint256 tokenId_ = createDefaultReceivable();
+        requestDefaultLoan(tokenId_);
+        _;
+    }
+
+    modifier whenDefaultLoanCreated() {
```
Comment (LouisTsai-Csie) [inline tests/integration/concrete/loan-manager/withdraw-funds/withdrawFunds.t.sol:93]: For readability, maybe renaming the modifier to `whenDefaultLoanFunded`.

**Review by LouisTsai-Csie — CHANGES_REQUESTED** [review]
There are some minor issues, please fix them, thanks

File: tests/Base.t.sol (hunk 1 above)
Comment (Jung0x12) [reply tests/Base.t.sol]: Agree, I'll update this

**Review by LouisTsai-Csie — COMMENTED** [review]
Which one do you use when you want to update the branch in comparison with main branch?

(1) git rebase
(2) git merge

I see there is a merge record in the commit log, maybe we should stay consistent.

**Comment by Jung0x12** [comment]
After discuss we decide to use `rebase`.

**Review by kevinsslin — APPROVED** [review]

**Pull request merged**